x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexCellIdSource.java

```diff
@@ -46,6 +52,7 @@ protected boolean advance(org.elasticsearch.common.geo.GeoPoint target) {
                 final long hex = H3.geoToH3(lat, lon, precision);
                 // validPoint is a fast check, validHex is slow
                 if (validPoint(lon, lat) || predicate.validHex(hex)) {
+                    assert predicate.validHex(hex) : H3.h3ToString(hex) + " should be valid but it is not";
```

Comment:
This check implies that if validPoint is true, then validHex must also be true, but then why call validHex in the `if` statement?

Reply:
I renamed the method to pointInBounds. The idea here is that checking if the point is inside the bounds is very fast and in that case we know that the hex must intersect the bounds too. Otherwise it might happen that the point is outside the bounds but the hex intersect the bounds, in that case the point is in the solution space too.

Reply:
Makes sense. I see now that this pattern is only to deal with the case where pointInBounds==false, but validHex is true, which can happen.